File: Minimum Size Subarray Sum.js
var minSubArrayLen = function(target, nums) {
  let sum = 0;
    let minLength = Infinity;
    let prevIndex = 0;
    
    for (let i = 0; i < nums.length; i++) {
        sum += nums[i];

        while (sum >= target) {
            minLength = Math.min(minLength, i - prevIndex + 1);
            sum -= nums[prevIndex];
            prevIndex++;
        }
    }
    
    
    return minLength === Infinity ? 0 : minLength;
}

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = right = 0
        currSum = nums[left]
        minLength = float('inf')
        while  right < len(nums):
            if currSum >= target:
                minLength = min(minLength, right - left + 1)
                currSum -= nums[left]
                left += 1
            else:
                right += 1
                if right < len(nums):
                    currSum += nums[right]
                        
        return minLength if minLength != float('inf') else 0
        